packages/pkcs12/src/__tests__/pkcs12-test.ts

```diff
@@ -0,0 +1,101 @@
+  password: 'password',
+  alias: 'test-alias',
+  md5Fingerprint: 'CCD9D0FD20A862B1E67EA838C99DD0C5',
+  sha1Fingerprint: '2EF9743A1BF12189AF5503787B54BFD3254B46A8',
+  sha256Fingerprint: 'EECDD35E90BB272D52EC261B4063C5D586980795452340525F9C8CC56DB53E2B',
+};
+describe('PKCS12 Tests', () => {
+  it('Conventional P12 Cert Fingerprints', async () => {
+    const {
+      base64EncodedP12,
+      password,
+      md5Fingerprint: expectedMd5Fingerprint,
+      sha1Fingerprint: expectedSha1Fingerprint,
+      sha256Fingerprint: expectedSha256Fingerprint,
+    } = conventionalP12;
+    const p12 = getPKCS12(base64EncodedP12, password);
+    const certificate = getX509Certificate(p12);
+    const md5Fingerprint = getCertificateFingerprint(certificate, {
+      upperCase: true,
+      hashAlgorithm: 'md5',
+    });
+    const sha1Fingerprint = getCertificateFingerprint(certificate, {
+      upperCase: true,
+      hashAlgorithm: 'sha1',
+    });
+    const sha256Fingerprint = getCertificateFingerprint(certificate, {
+      upperCase: true,
+      hashAlgorithm: 'sha256',
+    });
+    expect(md5Fingerprint).toEqual(expectedMd5Fingerprint);
+    expect(sha1Fingerprint).toEqual(expectedSha1Fingerprint);
+    expect(sha256Fingerprint).toEqual(expectedSha256Fingerprint);
+  });
+  it('Keystore P12 Cert Fingerprints', async () => {
+    const {
+      base64EncodedP12,
+      password,
+      alias,
+      md5Fingerprint: expectedMd5Fingerprint,
+      sha1Fingerprint: expectedSha1Fingerprint,
+      sha256Fingerprint: expectedSha256Fingerprint,
+    } = keystoreP12;
+    const p12 = getPKCS12(base64EncodedP12, password);
+    const certificate = getX509CertificateByFriendlyName(p12, alias);
+    const md5Fingerprint = getCertificateFingerprint(certificate, {
+      upperCase: true,
+      hashAlgorithm: 'md5',
+    });
+    const sha1Fingerprint = getCertificateFingerprint(certificate, {
+      upperCase: true,
+      hashAlgorithm: 'sha1',
+    });
+    const sha256Fingerprint = getCertificateFingerprint(certificate, {
+      upperCase: true,
+      hashAlgorithm: 'sha256',
+    });
+    expect(md5Fingerprint).toEqual(expectedMd5Fingerprint);
+    expect(sha1Fingerprint).toEqual(expectedSha1Fingerprint);
+    expect(sha256Fingerprint).toEqual(expectedSha256Fingerprint);
+  });
+  it('Conventional P12 Serial Number', async () => {
+    const { base64EncodedP12, password, serialNumber: expectedSerialNumber } = conventionalP12;
+    const p12 = getPKCS12(base64EncodedP12, password);
+    const certificate = getX509Certificate(p12);
+    const serialNumber = getFormattedSerialNumber(certificate);
+    expect(serialNumber).toEqual(expectedSerialNumber);
+  });
+  it('Conventional P12 Certificate Data', async () => {
+    const { base64EncodedP12, password } = conventionalP12;
+    const p12 = getPKCS12(base64EncodedP12, password);
+    const certificate = getX509Certificate(p12);
+    expect(certificate).toMatchSnapshot();
+  });
+});
```

Comment:
Consider adding tests for when getX509Certificate is called on a keystore, and when getX509CertificateByFriendlyName is called on a conventional .p12 file.